### PR TITLE
Add Slack template format option for Slack transport (Default & JSON)

### PR DESCRIPTION
**Pull request by ironick09**
This PR adds support for sending raw JSON payloads to slack in Alert templates. 

Changes:
- Adds new selection option to Slack Transport, `Slack Template Format`, set to `Default` by default, which is the existing behavior of the transport, and a selection `JSON` which sends the `$obj['msg']` as it is as the data payload to the Slack Incoming Webhook. 
  - When selecting JSON, the Alert Template should be formatted in valid JSON. You can still use Blade directives within this Template, but the parsed output needs to be valid JSON for the Slack API to actually post your message to a channel. 

This lets you build cool interactive/advanced messages using https://api.slack.com/tools/block-kit-builder and simply copying the resulting JSON into an Alert Template, substituting in the variables/values you want to print, and posting the message to slack. 

I did think about maybe validating the JSON if the format is set to JSON for the Slack transport, but I wasn't sure the best way to go about this, and how to return an error if the format is invalid. Any suggestions for this, or should I not worry about validating the JSON in `$obj['msg']`?

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

I am getting errors due to SNMP MIBs and uncaught `BindingResolutionException`, even when I clone from master and run `./scripts/pre-commit.php` before making any changes.

Errors:
```
E..........No log handling enabled - using stderr logging
MIB search path: /Users/jniec/.snmp/mibs:/usr/local/share/snmp/mibs:/usr/share/snmp/mibs
Cannot find module (SFA-INFO): At line 1 in (none)
Unknown object identifier: SFA-INFO::systemName.0

PHP Fatal error:  Uncaught Illuminate\Contracts\Container\BindingResolutionException: Target [Illuminate\Contracts\Debug\ExceptionHandler] is not instantiable. in /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php:960
Stack trace:
#0 /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(794): Illuminate\Container\Container->notInstantiable('Illuminate\\Cont...')
#1 /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(667): Illuminate\Container\Container->build('Illuminate\\Cont...')
#2 /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(615): Illuminate\Container\Container->resolve('Illuminate\\Cont...', Array)
#3 /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(767): Illuminate\Container\Container->make('Illuminate\\Cont...', Array)
#4 /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootst in /Users/jniec/source/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 960

Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 65536 bytes) in /Users/jniec/source/librenms/routes/console.php on line 51

```

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Alert/Transport/Slack.php [hunk 1]
```diff
@@ -39,10 +40,14 @@ public static function contactSlack($obj, $api)
...
-        $slack_msg     = strip_tags($obj['msg']);
-        $color         = self::getColorForState($obj['state']);
-        $data          = [
-            'attachments' => [
+
+        if ($api['fmt'] == 'JSON') {
+            $alert_message = $obj['msg'];
+        } else {
+            $slack_msg     = strip_tags($obj['msg']);
+            $color         = ($obj['state'] == 0 ? '#00FF00' : '#FF0000');
```
[the remark this answers is not in the document]
Comment (ironick09) [reply LibreNMS/Alert/Transport/Slack.php]: Sorry, fixed it to use `self::getcolorForState($obj['state']);`

**Comment by kkrumm1** [comment]
We need somebody to test this out

**Comment by SourceDoctor** [comment]
@ironick09 

Didn't get it to run

Alert Transport Slack changed to JSON.

Template Text:
```
{
	"blocks": [
		{
			"type": "section",
			"text": {
				"type": "mrkdwn",
				"text": "Hello, Assistant to the Regional Manager Dwight! *Michael Scott* wants to know where you'd like to take the Paper Company investors to dinner tonight.\n\n *Please select a restaurant:*"
			}
		}
                ]
}
```

nothing happens.

**Comment by laf** [comment]
@ironick09 Any update on the question above?

**Comment by ironick09** [comment]
Hey sorry, missed this, I will look into it today and report back.

**Pull request closed without merging**